### PR TITLE
Avoid auto-linkage using cmake [16415]

### PR DESCRIPTION
**Pull request by richiware**
## Description

Avoid auto-linkage in Windows through `#pragma` when the library was set through cmake.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- *N/A* Documentation builds and tests pass locally. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: fastrtps.repos [hunk 1]
```diff
@@ -19,3 +19,7 @@ repositories:
         type: git
         url: https://github.com/eProsima/IDL-Parser.git
         version: master
+    googletest-distribution:
+        type: git
+        url: https://github.com/google/googletest.git
+        version: release-1.11.0
```
Comment (EduPonz) [inline fastrtps.repos:25]: Please remove this

Comment (richiware) [reply fastrtps.repos]: Done in 27798485b

**Review by EduPonz — APPROVED** [review]

**Comment by EduPonz** [comment]
@richiware please test this

**Comment by EduPonz** [comment]
@richiprosima please test this

**Comment by EduPonz** [comment]
@richiprosima please test mac

**Comment by EduPonz** [comment]
@richiprosima please test mac

**Pull request merged**